frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +80,53 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and plural/kind
+ */
+export const useModelFinder = () => {
+  const models: ImmutableMap<string, K8sKind[]> = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models']),
+  );
+  const pluralsToModelMap = _.values(models.toJS()).reduce((acc, curr) => {
+    const ref = referenceForGroupVersionKind(curr.apiGroup)(curr.apiVersion)(curr.plural);
+    acc[ref] = curr;
+    return acc;
+  }, {});
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, kind: string) => {
+    if (!group) {
+      const ref = referenceForGroupVersionKind(group)('v1')(kind);
```

Comment:
Shouldn't this be...?

```suggestion
      const ref = referenceForGroupVersionKind('core')('v1')(kind);
```